### ISSUE TITLE
False Negative in SpotBugs Rule NN_NAKED_NOTIFY

### ISSUE THREAD

**Issue by clearlove2077** [spotbugs/spotbugs#3634]
Hi, I found a false negative bug about the rule NN_NAKED_NOTIFY. The minimized case is provided below. SpotBugs should have reported a warning at line 6 as the variable is visible to different threads and there are no modifications to mutable objects. This appears to be a false negative in the analysis.

```java
class Test {
    volatile String field = Test.class.getCanonicalName();
    static Test record = new Test();
    public static void foo() {
        synchronized (record.field) {
            record.field.notifyAll(); // should report a warning here, but no warnings
        }
    }
}
```

SpotBugs version: 4.9.3

**Comment by gtoison** [comment]
Thank you for reporting the issue, I've proposed a fix in #3635
